parquet/src/main/java/org/apache/iceberg/parquet/ApplyNameMapping.java

```diff
@@ -67,11 +67,17 @@ public Type list(GroupType list, Type elementType) {
     Preconditions.checkArgument(elementType != null,
         "List type must have element field");
 
+    boolean isOldListElementType = ParquetSchemaUtil.isOldListElementType(list);
     MappedField field = nameMapping.find(currentPath());
-    Type listType = Types.buildGroup(list.getRepetition())
-        .as(LogicalTypeAnnotation.listType())
-        .repeatedGroup().addFields(elementType).named(list.getFieldName(0))
-        .named(list.getName());
+
+    Types.GroupBuilder<GroupType> listBuilder = Types.buildGroup(list.getRepetition())
+        .as(LogicalTypeAnnotation.listType());
+    if (isOldListElementType) {
```

Comment:
I think this can check whether the repetition level of the element is `REPEATED` to determine whether it is a 2-level list. I think that would be better than allowing the `isOldListElementType` check in multiple places.